master/internal/user/postgres_users.go

```diff
@@ -137,15 +319,31 @@ func UserByToken(token string, ext *model.ExternalSessions) (
 	}
 
 	var user model.User
-	query := `
-	SELECT users.* FROM users
-	JOIN user_sessions ON user_sessions.user_id = users.id
-	WHERE user_sessions.id=?`
-
-	err = db.Bun().NewRaw(query, session.ID).Scan(context.Background(), &user)
+	err := db.Bun().NewSelect().
+		Table("users").
+		ColumnExpr("users.*").
+		Join("JOIN user_sessions").
+		JoinOn("user_sessions.user_id = users.id").
+		Where("user_sessions.id = ?", session.ID).Scan(ctx, &user)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return &user, &session, nil
 }
+
+// ByUsername looks up a user by name in the database.
+func ByUsername(ctx context.Context, username string) (*model.User, error) {
+	var user model.User
+	err := db.Bun().NewSelect().
+		Model(&user).
+		Where("username = ?", username).
+		Scan(context.Background())
+	if err != nil {
+		if errors.Cause(err) == sql.ErrNoRows {
+			return nil, db.ErrNotFound
+		}
+		return nil, err
```

Comment:
swap this to the standard switch style for multiple error checks and prefer errors.Is